Source/SwiftLintFramework/Rules/ModifierOrderRule.swift

```diff
@@ -183,7 +183,9 @@ public struct ModifierOrderRule: ASTRule, OptInRule, ConfigurationProviderRule {
         let filteredPreferedOrderOfModifiers = preferredOrderOfModifiers.filter(modifierGroupsInDeclaration.contains)
         for (index, preferredGroup) in filteredPreferedOrderOfModifiers.enumerated()
             where preferredGroup != modifierGroupsInDeclaration[index] {
-                return [StyleViolation(ruleDescription: type(of: self).description,
+                let reason = " \(preferredGroup) modifier should be before " +
+                             "\(modifierGroupsInDeclaration[index]) modifier."
+                return [StyleViolation(ruleDescription: type(of: self).description.appendDescription(with: reason),
```

Comment:
you don't need to change the description, `StyleViolation` has a `reason` parameter that can be used directly.